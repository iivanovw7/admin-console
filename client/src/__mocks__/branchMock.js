export const BRANCH = {
  address: '938 Barby Way',
  email: 'mbrazier0@amazon.co.uk',
  fax: '+86 829 425 7497',
  information: '1',
  name: 'San Mart√≠n de los Andes',
  phone: '+54 275 474 9318',
  status: true,
  __v: 0,
  _id: '5c8f96b9cece9911f01dd092'
};