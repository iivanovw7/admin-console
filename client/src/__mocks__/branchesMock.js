export const BRANCHES = {
  page: 1,
  limit: 7,
  pages: 4,
  results: 25,
  output: [
    {
      address: '938 Barby Way',
      email: 'mbrazier0@amazon.co.uk',
      fax: '+86 829 425 7497',
      information: '1',
      name: 'San Mart√≠n de los Andes',
      phone: '+54 275 474 9318',
      status: true,
      __v: 0,
      _id: '5c8f96b9cece9911f01dd092'
    },
    {
      address: '903 Dottie Street',
      email: 'inerger1@businesswire.com',
      fax: '+86 230 609 6364',
      information: 't',
      name: 'Dingzhai',
      phone: '+86 103 802 9655',
      status: true,
      __v: 0,
      _id: '5c8f96b9cece9911f01dd093'
    },
    {
      status:false,
      _id:"5c8f96b9cece9911f01dd0a0",
      name:"Cantao-an",
      email:"atroye@dailymail.co.uk",
      phone:"+63 269 360 0515",
      fax:"+86 713 306 1827",
      address:"7827 Luster Point",
    }
  ]
};
