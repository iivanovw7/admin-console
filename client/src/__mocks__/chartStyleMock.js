export const style = {
  type: "PieChart",
  id: 2
};
