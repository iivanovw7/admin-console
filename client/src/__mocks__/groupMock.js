export const GROUP = {
  status: true,
  permissions: true,
  _id: '5c8f96bacece9911f01dd0a6',
  name: 'Front-End Pool',
  description: 'Group of all employees with Front-End expertise'
};