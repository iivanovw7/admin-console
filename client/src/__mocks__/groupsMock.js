export const GROUPS = {
  page: 1,
  limit: 7,
  pages: 4,
  results: 25,
  output: [
    {
      status: true,
      permissions: true,
      _id: '5c8f96bacece9911f01dd0a6',
      name: 'Front-End Pool',
      description: 'Group of all employees with Front-End expertise',

    },
    {
      status: false,
      permissions: false,
      _id: '5c8f96bacece9911f01dd0a7',
      name: 'Back-End Pool',
      description: 'Group of all employees with Back-End expertise',
    },
    {
      status:false,
      permissions:false,
      _id:"5cade7a578f0a97097c972c5",
      name:"TEST",
      description:"TEST",
    }
  ]
};
