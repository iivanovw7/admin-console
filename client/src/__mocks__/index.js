export * from './branchesMock';
export * from './branchMock';
export * from './userMock';
export * from './usersMock';