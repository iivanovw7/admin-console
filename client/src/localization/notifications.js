import * as types from '../constants/ActionTypes';

export function formLocalizedNotification(name, action, locale, newState) {

    switch (action.type) {
      case types.ERROR: {
        return `${strings[locale].error}`;
      }
      case types.FETCH_ERROR: {
        return `${strings[locale].fetchError}`;
      }
      case types.AUTHENTICATION_ERROR: {
        return `${strings[locale].authError}`;
      }
      case types.UPDATE_BRANCH: {
        return `${name} ${strings[locale].updateSuccess}`;
      }
      case types.UPDATE_GROUP: {
        return `${name} ${strings[locale].updateSuccess}`;
      }
      case types.UPDATE_ROLE: {
        return `${name} ${strings[locale].updateSuccess}`;
      }
      case types.UPDATE_USER: {
        return `${name} ${strings[locale].updateSuccess}`;
      }
      case types.ADD_BRANCH: {
        return `${name} ${strings[locale].addSuccess}`;
      }
      case types.ADD_GROUP: {
        return `${name} ${strings[locale].addSuccess}`;
      }
      case types.ADD_ROLE: {
        return `${name} ${strings[locale].addSuccess}`;
      }
      case types.CHANGE_GROUP_STATUS: {
        return `${name} ${strings[locale].changeStatus} "${newState}"!`;
      }
      case types.CHANGE_ROLE_STATUS: {
        return `${name} ${strings[locale].changeStatus} "${newState}"!`;
      }
      case types.CHANGE_USER_STATUS: {
        return `${name} ${strings[locale].changeStatus} "${newState}"!`;
      }
      case types.DELETE_GROUP: {
        return `${name} ${strings[locale].deleted}`;
      }
      case types.DELETE_ROLE: {
        return `${name} ${strings[locale].deleted}`;
      }
      default:
        return '';
    }
}

let strings = {
  en: {
    error: 'Error!',
    fetchError: 'Error while getting data!',
    authError: 'Authentication error!',
    addSuccess: 'created!',
    updateSuccess: 'successfully modified!',
    changeStatus: 'status changed to',
    deleted: 'successfully removed!'
  }

};


