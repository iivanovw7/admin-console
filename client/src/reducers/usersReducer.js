import * as types from '../constants/actionTypes';
import { formLocalizedNotification as message }  from '../localization/notifications';

const initialState = {
  list: {
    page: 1,
    search: null,
  },
  user: null,
  error: null,
  success: null,
  confirmed: false,
};

function displayStatus(status) {
  return status ? 'Active' : 'Disabled';
}

export default function (state = initialState, action) {

  switch (action.type) {
    case types.FETCH_USERS: {
      return {
        ...state,
        list: action.payload.data,
        error: null,
        success: null
      };
    }
    case types.SEARCH_USERS: {
      return {
        ...state,
        list: action.payload.data,
        error: null,
        success: null
      };
    }
    case types.FETCH_USER: {
      return {
        ...state,
        user: action.payload.data,
        error: null,
        success: null
      };
    }
    case types.UPDATE_USER: {
      return {
        ...state,
        user: {},
        error: null,
        success: message(
          action.payload.data.name + ' ' + action.payload.data.surname,
          action,
          'en'
        ),
        confirmed: false,
      };
    }
    case types.CHANGE_USER_STATUS: {
      return {
        ...state,
        error: null,
        success: message(
          action.payload.data.name + ' ' + action.payload.data.surname,
          action,
          'en',
          displayStatus(action.payload.data.active)
        ),
        confirmed: false,
      };
    }
    case types.ERROR: {
      return {
        ...state,
        list: {
          page: 1,
          search: null,
        },
        user: null,
        error: message(
          null,
          action,
          'en'
        ),
        success: null,
        confirmed: false,
      };
    }
    case types.FETCH_ERROR: {
      return {
        ...state,
        list: {
          page: 1,
          search: null,
        },
        user: null,
        error: message(
          null,
          action,
          'en'
        ),
        success: null,
        confirmed: false,
      };
    }
    case types.CONFIRM_NOTIFICATION: {
      return {
        ...state,
        confirmed: true,
      }
    }
    default:
      return {
        ...state
      };
  }

}
