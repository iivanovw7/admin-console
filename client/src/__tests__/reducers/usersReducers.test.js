import * as types from '../../constants/actionTypes';
import { formLocalizedNotification as message } from '../../localization/notifications';
import branchesReducer from '../../reducers/branchesReducer';
import usersReducer, { displayStatus } from '../../reducers/usersReducer';
import * as mocks from './../../__mocks__/';

const setup = () => {

  describe('USERS reducers', () => {

    it('FETCH USERS', () => {
      const expectedActions = {
        type: types.FETCH_USERS,
        payload: {
          data: mocks.USERS
        }
      };

      expect(usersReducer({}, expectedActions)).toEqual({
        list: expectedActions.payload.data,
        error: null,
        success: null
      });

    });

    it('FETCH USER', () => {
      const expectedActions = {
        type: types.FETCH_USER,
        payload: {
          user: mocks.USER
        }
      };

      expect(usersReducer({}, expectedActions)).toEqual({
        user: expectedActions.payload.data,
        error: null,
        success: null
      });

    });

    it('UPDATE USER', () => {
      const expectedActions = {
        type: types.UPDATE_USER,
        payload: {
          data: mocks.USER
        }
      };

      expect(usersReducer({}, expectedActions)).toEqual({
        user: {},
        error: null,
        success: message(
          expectedActions.payload.data.name + ' ' + expectedActions.payload.data.surname,
          expectedActions,
          'en'
        ),
        confirmed: false
      });

    });

    it('CHANGE USER STATUS', () => {
      const expectedActions = {
        type: types.CHANGE_USER_STATUS,
        payload: {
          data: mocks.USER
        }
      };

      expect(usersReducer({}, expectedActions)).toEqual({
        error: null,
        success: message(
          expectedActions.payload.data.name + ' ' + expectedActions.payload.data.surname,
          expectedActions,
          'en',
          displayStatus(expectedActions.payload.data.active)
        ),
        confirmed: false
      });
    });

    it('GENERAL ERROR', () => {
      const expectedActions = {
        type: types.GENERAL_ERROR,
        payload: {
          data: mocks.USER
        }
      };

      expect(usersReducer({}, expectedActions)).toEqual({
        list: {
          page: 1,
          search: null
        },
        user: null,
        error: message(
          null,
          expectedActions,
          'en'
        ),
        success: null,
        confirmed: false
      });
    });

    it('FETCH ERROR', () => {
      const expectedActions = {
        type: types.FETCH_ERROR,
        payload: {
          data: mocks.USER
        }
      };

      expect(usersReducer({}, expectedActions)).toEqual({
        list: {
          page: 1,
          search: null
        },
        user: null,
        error: message(
          null,
          expectedActions,
          'en'
        ),
        success: null,
        confirmed: false
      });
    });

    it('CONFIRM NOTIFICATION', () => {
      const expectedActions = {
        type: types.CONFIRM_NOTIFICATION,
        confirmed: true
      };

      expect(branchesReducer({}, expectedActions)).toMatchSnapshot();
    });

  });

};

setup();