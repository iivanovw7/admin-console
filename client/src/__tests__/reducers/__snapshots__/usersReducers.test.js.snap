// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`USERS reducers CONFIRM NOTIFICATION 1`] = `
Object {
  "confirmed": true,
}
`;
